mls_validation_service/src/handlers.rs

```diff
@@ -158,6 +198,43 @@ async fn validate_inbox_id_key_package(
     })
 }
 
+async fn verify_smart_contract_wallet_signatures(
+    signatures: Vec<UnverifiedSmartContractWalletSignature>,
+    scw_verifier: &dyn SmartContractSignatureVerifier,
+) -> Result<Response<VerifySmartContractWalletSignaturesResponse>, Status> {
+    let mut responses = vec![];
+    for request in signatures {
+        let Some(signature) = request.scw_signature else {
```

Comment:
If the client is expecting to zip together the requests and the responses by index then we should insert a `false` here rather than skipping the entry. Otherwise the request at index 1 might get matched with the response at index 2.

Reply:
Do you know why this field is optional at all? It's not declared as optional in the protos, but comes in as optional in the generated spec. https://github.com/xmtp/proto/pull/211/files#diff-a8714baf14c9f3a47e527cf66f630eb43d2e063fc68413d73e8ee5bf561fc44eR62

Reply:
Got it, that's an interesting limitation. I was aware of the mismatched index issue, and was perplexed about that being generated as optional when the spec wasn't marked as such, and meant to bring it up. I'll switch this logic out to put in a false where there's a `None`.

Reply:
Updated